<b>Исследование данных о пользователях</b>  
<u>Обзор и подготовка данных</u>

In [ ]:
import pandas as pd
import numpy as np
user_data = pd.read_csv(r'C:\Users\user\Desktop\practice\3_user_data.csv')
logs = pd.read_csv(r'C:\Users\user\Desktop\practice\3_logs.csv')
print('Пользователи:\n', user_data.head())
print('Логи:\n', logs.head())

Пользователи:
    client  premium  age
0   46346    False   58
1    4391    False   55
2   27372    False   64
3   11989    False   44
4   60664    False   49
Логи:
    client  success  platform        time
0   40177     True     phone  1585412212
1   61468     True     phone  1585425658
2   35604    False     phone  1585459894
3    9961     True     phone  1585400189
4   10931     True  computer  1585468629


Можно сразу заметить, что столбец time имеет не самый удобный для обработки формат. Далее рассмотрим основные характеристики датасетов.

In [4]:
print(user_data.info(),'\n')
print(logs.info())

<class 'pandas.DataFrame'>
RangeIndex: 2954 entries, 0 to 2953
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype
---  ------   --------------  -----
 0   client   2954 non-null   int64
 1   premium  2954 non-null   bool 
 2   age      2954 non-null   int64
dtypes: bool(1), int64(2)
memory usage: 49.2 KB
None 

<class 'pandas.DataFrame'>
RangeIndex: 4500 entries, 0 to 4499
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype
---  ------    --------------  -----
 0   client    4500 non-null   int64
 1   success   4500 non-null   bool 
 2   platform  4500 non-null   str  
 3   time      4500 non-null   int64
dtypes: bool(1), int64(2), str(1)
memory usage: 110.0 KB
None


<b>Выводы</b>  
Всего в таблице user_data 3 столбца и 2954 строки, типы данных - int64, bool.  
В таблице logs - 4 столбца и 4500 строк, типы данных - int64, bool, str.  
Описание данных:  
user_data:  
• client – идентификатор пользователя  
• premium – является ли клиент премиум  
• age – возраст  
logs:  
• client – идентификатор пользователя  
• success – результат (успех - 1, нет - 0)  
• platform – платформа  
• time – время в формате Unix  

В обеих таблицах отсутствуют пропуски в данных.

In [ ]:
logs.platform.unique() # уникальные значения в столбце platform
succ = logs.groupby(['client'], as_index=False) \
            .agg({'success': 'sum'}) \
            .sort_values(by = 'success', ascending=False)
max_succ = succ['success'].max() # находим значение максимального кол-ва успешных операций
succ = succ[succ.success == max_succ].client.tolist() #запись id клиентов с max кол-вом успешных операций в виде списка
print(succ)

[12179, 36165, 28719, 61473, 92584, 82563, 78349, 52870, 61468]
